### DISCUSSION TITLE
What is the current state of Consumer Pull?

### DISCUSSION THREAD

**Discussion by reisman234** [eclipse-edc/Connector#1361] [Q&A] [answered]
I found in the documentation some information about different transfer methods, `provider push` and `consumer pull` in the [data-plane-framework](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/main/docs/developer/data-plane-framework/dpf_transfer.md). If I'm right, only the `provider push` is shown in the samples. 

If  I'm reading the documentation right, you do not support this method right now, but the extension exits anyway in the repository?

 - is it usable?
 - could you provide some resources how to use it (sample or documentation)
 - any plans for when it will be supported?

**Comment by ndr-brt** [comment, marked as answer]
The `consumer pull` method is supported, it's described by the `end-to-end-transfer-test`:
https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/main/system-tests/e2e-transfer-test/runner/src/test/java/org/eclipse/dataspaceconnector/test/e2e/AbstractEndToEndTransfer.java#L38

A sample that show this (and also `provider push` as well) could be created (or sample-04.0 could be leveraged for this)

**Reply by reisman234** [reply]
I invested some time in this `system-tests/e2e-transfer-test` and my first approach was to extend the `samples/04.0-file-transfer` sample with the required extension and configuration settings, but in the end this does not work.

So it would be nice to have a documentation with the minimal requirements  to set up a connector for `consumer pull`
